Makefile-legacy.fedora

```diff
@@ -0,0 +1,91 @@
+# This file is included from Makefile.fedora if package do not define RPM_USE_MOCKBUILD
+
+RPM_BUILD_DEFINES += --define "_rpmdir $(OUTPUT_DIR)/"
+RPM_BUILD_DEFINES += --define "qubes_builder 1"
+RPM_BUILD_DEFINES += --define "_sourcedir $(DIST_SRC)"
```

Comment:
It is in the head of Makefile.fedora for which Makefile-legacy.fedora is included

Reply:
But it shouldn't be used for mock-based build, because `@BACKEND_VMM@`.

Reply:
not sure not understand: the variable backend_vmm is defined using RPM_BUILD_DEFINES for both of the Makefile-legacy.fedora and Makefile-mock.fedora. And in both cases, the @BACKEND_VMM@ is replaced. What do I miss?

Reply:
backend_vmm shouldn't be needed/used in RPM_BUILD_DEFINES for Makefile-mock.fedora. All qubes-specific variables should be embedded at generate-spec step.

Reply:
Oh ok. So your remark was to remove backend_vmm not in Makefile-legacy.fedora but for Makefile.fedora ?

Reply:
Yes, exactly. I should have worded it better.

Reply:
But it is needed in Makefile-legacy.fedora...

Reply:
Hum I think I understand what is wrong.

Reply:
I should not have removed it at all. I don't see why mock does not need it whereas the legacy builder does. E.g. in core-libvirt it is used as part of rpm names where it also used to include requires. In both cases with either Mock or the Legacy builder, I need to replace @BACKEND_VMM@ and also to have access to the value of %{backend_vmm}. I'm sorry if I really don't understand...

Reply:
`--define backend_vmm ...` is Qubes-specific rpmbuild option to specify what "virtual machine monitor" should be used ([details](http://theinvisiblethings.blogspot.de/2013/03/introducing-qubes-odyssey-framework.html)). For now only `xen` is supported. But we can add `kvm` or such.
It should be possible to build src.rpm without it, with actual value embedded at spec generation time.
This means `@BACKEND_VMM@` should _replace_ usage of `%{backend_vmm}` in spec.in. And that `--define` shouldn't be needed in mock-based build.
But we still need it for legacy build.

> E.g. in core-libvirt it is used as part of rpm names where it also used to include requires.

The only place where `%{backend_vmm}` is used in core-libvirt is a comment...

Reply:
My bad it was core-admin (https://github.com/QubesOS/qubes-core-admin/blob/3c2ac0fd414afbee2ce8737a80f53ad17f12c7c8/rpm_spec/core-dom0.spec#L88) but in fact something like ```%{!?backend_vmm: %define backend_vmm @BACKEND_VMM@}``` could be added.

Reply:
In fact the macro is still only needed for legacy builds (not the legacy method) this is what you meant and what I was not thinking about?

Reply:
I'm not sure about distinction "legacy builds"/"legacy method". The macro from your comment above may be useful as a method for migration (just add this one into .spec.in instead of replacing every `%{backend_vmm}` instance). But the final effect (after rpm parsing the spec) should be the same as replacing `%{backend_vmm}` with `@BACKEND_VMM@`. I don't have preference which method you use.

Old method (R3.2 packages) do require `--define backend_vmm ...`.

Reply:
I will try to be more explicit. We have been talking around because I was not enough explicit. Sorry. I was thinking, legacy method as the current Qubes method of building things and by legacy builds I was thinking about build for R3.2 packages...This is why I was finally confused...

Reply:
Looks like you've restored backend_vmm in `Makefile.fedora` (in addition to adding it in `Makefile-legacy.fedora`), was this intentional?

Reply:
No...corrected.